File: rust/src/models/environment_in.rs
// this file is @generated
use serde::{
    Deserialize,
    Serialize,
};

use super::{
    connector_in::ConnectorIn,
    event_type_in::EventTypeIn,
};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct EnvironmentIn {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub connectors: Option<Vec<ConnectorIn>>,

    #[serde(rename = "eventTypes")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub event_types: Option<Vec<EventTypeIn>>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub settings: Option<serde_json::Value>,
}

impl EnvironmentIn {
    pub fn new() -> Self {
        Self {
            connectors: None,
            event_types: None,
            settings: None,
        }
    }
}

File: python/svix/models/environment_in.py
# this file is @generated
import typing as t

from .common import BaseModel
from .connector_in import ConnectorIn
from .event_type_in import EventTypeIn


class EnvironmentIn(BaseModel):
    event_types: t.Optional[t.List[EventTypeIn]] = None

    settings: t.Optional[t.Dict[str, t.Any]] = None

    connectors: t.Optional[t.List[ConnectorIn]] = None
